jasmine/spec/hoochSpec.js

```diff
@@ -331,6 +331,102 @@ describe("hooch", function() {
       expect(form.submit).not.toHaveBeenCalled();
     })
   })
+  describe('IhHistoryState', function(){
+    it('toQueryString', function(){
+      ih_history_state = new hooch.IhHistoryState({skeleton_key: 'treasure', foobar: 'baz'})
+      var query_string = ih_history_state.toQueryString()
+      expect(query_string).toEqual('skeleton_key=treasure&foobar=baz')
+    })
+    it('toUrl', function(){
+      ih_history_state = new hooch.IhHistoryState({skeleton_key: 'treasure', foobar: 'baz'})
+      var url = ih_history_state.toUrl()
+      expect(url).toEqual([location.protocol,'//',location.host,location.pathname,'?','skeleton_key=treasure&foobar=baz'].join(''))
+    })
+    it('addState', function(){
+      ih_history_state = new hooch.IhHistoryState({foobar: 'baz'})
+      ih_history_state.addState('skeleton_key','treasure')
+      var state = ih_history_state.state
+      expect(state.foobar).toEqual('baz') // We didn't mess with the original state
+      expect(state.skeleton_key).toEqual('treasure') // But we did add to it
+    })
+    it('addPath', function(){
+      ih_history_state = new hooch.IhHistoryState({skeleton_key: 'treasure', foobar: 'baz'})
+      ih_history_state.addPath('/my/new/path')
+      var new_path = ih_history_state.new_path
+      expect(new_path).toEqual('/my/new/path')
+    })
+    it('newPath', function(){
+      ih_history_state = new hooch.IhHistoryState({skeleton_key: 'treasure', foobar: 'baz'})
+      ih_history_state.addPath('/my/new/path')
+      expect(ih_history_state.newPath()).toEqual([location.protocol,'//',location.host,'/my/new/path'].join(''))
+    })
+    it('addKeyValue', function(){
+      ih_history_state = new hooch.IhHistoryState({skeleton_key: 'treasure', foobar: 'baz'})
+      history = {
+        state: {},
+        replaceState: function(state, throwaway, newpath){
+          new_state = state
+          new_path = newpath
+        }
+      }
+      ih_history_state.addKeyValue('my_key','my_value')
+      expect(new_state.my_key).toEqual('my_value')
+      expect(new_path).toEqual([location.protocol,'//',location.host,location.pathname,'?','skeleton_key=treasure&foobar=baz&my_key=my_value'].join(''))
+      delete history
```

Comment:
@gomayonqui I created history as a global variable to mock the window.history object for the test. But since it is global, I delete it so it won't affect other tests.